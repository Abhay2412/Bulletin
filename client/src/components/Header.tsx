// IMPORTING APIS
import React from "react";
import { AppBar, Toolbar, IconButton, useMediaQuery, Button, Menu, MenuItem, ListItemIcon} from "@material-ui/core";
import Typography from '@mui/material/Typography';
import MenuIcon from '@mui/icons-material/Menu';
import { makeStyles, useTheme } from "@material-ui/core/styles";
import { BrowserRouter, Route, Routes, Link } from "react-router-dom";
// REACT APP IMPORTS
import LogIn from "../pages/Login";
import SignIn from "../pages/Signin";
//The local styling used here 
const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  title: {
    flexGrow: 1
  },
  tabs: {
    width: '100%'
  }
}));

const Header = () => {
  const classes = useStyles();
  const [anchor, setAnchor] = React.useState(null);
  const open = Boolean(anchor);
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
  const [anchorElNav, setAnchorElNav] = React.useState<null | HTMLElement>(null);
  const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElNav(event.currentTarget);
  };

  return (
    <div className={classes.root}>
      <BrowserRouter>
          <AppBar position="static"  style={{ background: '#F0E6DB' }}>
            <Toolbar>
              <Typography
                variant="h6"
                noWrap
                component="div"
                className={classes.title}
                sx={{ mr: 2, color: '#534029', display: { xs: 'none', md: 'flex' } }}
              >
                BULLETIN
              </Typography>
              {isMobile ? (
                <>
                  <IconButton
                    color="inherit"
                    className={classes.menuButton}
                    edge="start"
                    aria-label="menu"
                    onClick={handleMenu}
                  >
                    <MenuIcon />
                  </IconButton>
                  <Menu
                    id="menu-appbar"
                    anchorEl={anchor}
                    anchorOrigin={{
                      vertical: "top",
                      horizontal: "right"
                    }}
                    transformOrigin={{
                      vertical: "top",
                      horizontal: "right"
                    }}
                    open={open}
                  >
                    <MenuItem
                      onClick={() => setAnchor(null)}
                      component={Link}
                      to="/login"
                    >
                      <Typography variant="h6">Main</Typography>
                    </MenuItem>
                    <MenuItem
                      onClick={() => setAnchor(null)}
                      component={Link}
                      to="/login"
                    >
                      <Typography variant="h6">Log In</Typography>
                    </MenuItem>
                    <MenuItem
                      onClick={() => setAnchor(null)}
                      component={Link}
                      to="/login"
                    >
                      <Typography variant="h6">Sign Up</Typography>
                    </MenuItem>
                  </Menu>
                </>
              ) : (
                <div style={{ marginRight: "2rem" }}>
                  <Button
                    variant="text"
                    color="default"
                    component={Link}
                    to="/login"
                  >
                    Main
                  </Button>
                  <Button
                    variant="text"
                    color="default"
                    component={Link}
                    to="/login"
                  >
                    Log In
                  </Button>
                  <Button
                    variant="text"
                    color="default"
                    component={Link}
                    to="/login"
                  >
                   Sign Up
                  </Button>
                </div>
              )}
            </Toolbar>
          </AppBar>
          <Routes>
            <Route path="/login" element={LogIn} />
            <Route path="/signin" element={SignIn} />
          </Routes>
          </BrowserRouter>
    </div>
  );
};

export default Header;