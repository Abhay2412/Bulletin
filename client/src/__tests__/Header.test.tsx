import React from "react";
import { configure, mount, ReactWrapper, shallow } from "enzyme";
import { AppBar, Toolbar, Button, IconButton } from "@material-ui/core";
import Header from "../components/Header";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

describe("testing <Header /> component", () => {
  const wrapper: ReactWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>
  );

  it("should render properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Header />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render one Header component", () => {
    expect(wrapper.find(AppBar).length).toEqual(1);
  });

  it("should render header with static poistioning", () => {
    expect(wrapper.find(AppBar).prop("position")).toEqual("static");
  });

  it("should render exactly three <Button /> component", () => {
    expect(wrapper.find(Button).length).toEqual(3);
  });

  it("should render exactly 1 <Toolbar /> component", () => {
    expect(wrapper.find(Toolbar).length).toEqual(1);
  });
});
