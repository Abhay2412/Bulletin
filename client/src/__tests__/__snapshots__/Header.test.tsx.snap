// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Header /> component should render properly 1`] = `<div />`;
